src/actions/messageActions.js

```diff
@@ -0,0 +1,44 @@
+import firebaseApi from '../api/firebase';
+import * as types from './actionTypes';
+
+import {ajaxCallError, beginAjaxCall} from './ajaxStatusActions';
+
+export function sendMessageInCurrentRoom(message) {
+  return (dispatch, getState) => {
+    const currentRoomId = getState().rooms.current;
+    const newMessage = {
+      sender: getState().user.uid,
+      sender_name: getState().user.email,
+      content: message,
+      date: firebaseApi.TIMESTAMP_CONST
+    };
+    return firebaseApi.databasePush('/messages/'+currentRoomId, newMessage)
+      .then( () => {
+        dispatch(messageSentSuccess());
+      })
+      .catch( error => {
+        dispatch(ajaxCallError(error));
+        // @TODO better error handling
```

Comment:
Do you have ideas of what you'd like to expect from a user's POV? And how to implement it? Firestore might have some tricks for that.

Inspiration: https://uxplanet.org/optimistic-1000-34d9eefe4c05